gsonpath-compiler/src/main/java/gsonpath/model/GsonObjectFactory.kt

```diff
@@ -95,24 +91,25 @@ class GsonObjectFactory(private val subTypeMetadataFactory: SubTypeMetadataFacto
 
         val gsonSubTypeMetadata = subTypeMetadataFactory.getGsonSubType(fieldInfo)
 
-        if (jsonFieldPath.contains(flattenDelimiter.toString())) {
-            addNestedType(gsonPathObject, fieldInfo, jsonFieldPath, flattenDelimiter, fieldInfoIndex, isRequired,
-                    fieldName, gsonSubTypeMetadata)
+        if (jsonFieldPath.contains(metadata.flattenDelimiter.toString())) {
+            addNestedType(gsonPathObject, fieldInfo, jsonFieldPath, metadata.flattenDelimiter, fieldInfoIndex,
+                    isRequired, fieldName, gsonSubTypeMetadata)
 
         } else {
             addStandardType(gsonPathObject, fieldInfo, jsonFieldPath, fieldInfoIndex, isRequired, gsonSubTypeMetadata)
         }
     }
 
     @Throws(ProcessingException::class)
-    private fun addNestedType(gsonPathObject: GsonObject,
-                              fieldInfo: FieldInfo,
-                              initialJsonFieldPath: String,
-                              flattenDelimiter: Char,
-                              fieldInfoIndex: Int,
-                              isRequired: Boolean,
-                              fieldName: String,
-                              gsonSubTypeMetadata: SubTypeMetadata?) {
+    private fun addNestedType(
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 23 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=kotlin%3AS3776)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This function has 8 parameters, which is greater than the 7 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=kotlin%3AS107)